### PR TITLE
enable --output-format json everywhere appropriate

### PR DESCRIPTION
**Pull request by csells**
### Related Issue

**Issue:** #6996

### Description

This PR implements complete JSON output support for the Cline CLI, making all batch commands fully scriptable and integration-friendly when using the `--output-format json` flag (aka `-F json`).

**Problem**
Currently, JSON output is only partially implemented for streaming conversation messages. Most CLI commands (instance management, config, logs, version) lack structured JSON output. Furthermore, even the task-related commands that do support JSON output do so with mixed text and JSON. All of this makes it difficult to:
- Build automation scripts around Cline CLI
- Integrate Cline into CI/CD pipelines
- Parse command results programmatically
- Monitor Cline operations via machine-readable output

**Solution**
This PR implements support for JSON output across all rest of the CLI commands while maintaining backward compatibility with existing JSON, plain and rich text formats.

**Key Features:**
- ✅ **Universal JSONL output** - All commands output valid JSON Lines format
- ✅ **Verbose composition** - `--verbose` flag produces structured debug messages in JSON mode
- ✅ **Zero mixed text/JSON** - Strict JSON-only output in JSON mode
- ✅ **Interactive command rejection** - `auth` and `task chat` properly reject JSON mode with clear plain text errors
- ✅ **Backward compatibility** - Existing plain/rich outputs preserved
- ✅ **Consistent structure** - All batch responses follow standard format with type, command, status, and data fields

**Implementation Approach:**
 The implementation follows a dual-format architecture:
1. **Format 1 (Streaming)**: Preserved existing `{type:"say"|"ask"}` format for task conversations
2. **Format 2 (Batch)**: New `{type:"command", status:"success"|"error"|"debug"}` format for all other commands

Notes:
- Added helper functions for format detection (`JsonFormat()`, `PlainFormat()`, `RichFormat()`)
- Created centralized JSON output utilities in `pkg/cli/output/json.go`
- Applied consistent if/else pattern across all commands to add JSON branches while preserving original output
- Added PreRunE hooks for interactive commands to reject JSON mode appropriately
- Ensure that verbose output streams as JSONL in JSON mode along with the output
- Ensure that errors in JSON mode are output as JSON
 
### Test Procedure

**Testing Philosophy:**
Testing was done across multiple dimensions to catch any edge cases.

**Test Coverage**

1. **Format Validation Tests**
   - JSON validity for all 23 commands
   - Structure verification (required fields present)
   - Data completeness checks
   - Zero text leakage verification

2. **Backward Compatibility Tests**
   - Plain format: Verified output unchanged from merge base
   - Rich format: Verified output unchanged from merge base
   - Compared line-by-line with original outputs
   - Git diff verification shows only additive changes

3. **Cross-Format Tests**
   - 3 formats × 3 command types × 2 states × 2 verbosity = 36 combinations
   - Batch commands (19 commands)
   - Interactive commands (2 commands - proper rejection verified)
   - Streaming commands (2 commands - Format 1 preserved)

4. **Verbose Mode Tests**
   - JSON + verbose produces JSONL debug messages
   - Plain + verbose produces original text
   - Rich + verbose produces original formatted output
   - Example: `instance new --verbose -F json` outputs ~22 debug lines

5. **Error Handling Tests**
   - Errors formatted as JSON in JSON mode
   - Plain text errors in interactive command rejection
   - Network errors, missing parameters, invalid inputs

**NOTE**: Some of the tests that create a new instance or database are flakey when run as part of the suite. These tests pass in isolation.

### Type of Change

- [ ]  Bug fix (non-breaking change which fixes an issue)
- [x]  New feature (non-breaking change which adds functionality)
- [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ]  Refactor Changes
- [ ]  Cosmetic Changes
- [ ]  Documentation update
- [ ]  Workflow Changes

### Pre-flight Checklist

- [x] Changes are limited to a single feature, bugfix or chore (split larger changes into separate PRs)
- [x] Tests are passing (`cd cli && go test ./e2e/... -v`) and code is formatted and linted (`go fmt`)
- [ ] I have created a changeset using `npm run changeset` (required for user-facing changes)
- [x] I have reviewed [contributor guidelines](https://github.com/cline/cline/blob/main/CONTRIBUTING.md)

### Output Examples

**1. Simple Command (version):**
```bash
$ cline version -F json
{"type":"command","command":"version","status":"success","data":{"cliVersion":"dev","coreVersion":"dev","commit":"unknown","date":"unknown","builtBy":"unknown","goVersion":"go1.25.3","os":"darwin","arch":"arm64"}}
```

**2. Verbose Mode (~22 debug lines):**
```bash
$ cline instance new --verbose -F json
{"type":"command","command":"instance new","status":"debug","message":"Starting new Cline instance..."}
{"type":"command","command":"cline","status":"debug","message":"Starting cline-host on port 59030"}
{"type":"command","command":"cline","status":"debug","message":"Started cline-host (PID: 18191)"}
... (19 more debug lines) ...
{"type":"command","command":"instance new","status":"success","data":{"address":"127.0.0.1:59029","corePort":59029,"hostPort":59030,"isDefault":false}}
```

**3. Error Response:**
```bash
$ cline instance kill nonexistent:9999 -F json
{"command":"instance kill","error":"instance nonexistent:9999 not found in registry","status":"error"}
```

**4. Interactive Rejection:**
```bash
$ cline auth -F json
Error: auth is an interactive command and cannot be used with --output-format json
```

**5. Complex Command (instance list):**
```bash
$ cline instance list -F json
{"type":"command","command":"instance list","status":"success","data":{"defaultInstance":"127.0.0.1:58973","instances":[{"address":"127.0.0.1:58973","status":"SERVING","version":"3.34.0","lastSeen":"2025-10-26T19:52:01-07:00","pid":"18175","platform":"CLI","isDefault":true}]}}
```

----

> [!IMPORTANT]
> This PR enables JSON output across all CLI commands, enhancing scriptability and consistency, with comprehensive error handling and verbose mode support.
> 
>   - **Behavior**:
>     - Adds JSON output support to all CLI commands using `--output-format json`.
>     - Ensures interactive commands reject JSON mode with plain text errors.
>     - Implements JSONL format for verbose output.
>   - **Output**:
>     - Introduces `output/json.go` for JSON output utilities.
>     - Updates `version`, `instance`, `logs`, `config`, and `task` commands to use JSON output.
>     - Adds tests in `output/json_test.go` for JSON response formatting.
>   - **Commands**:
>     - Modifies `newInstanceCommand`, `newLogsCommand`, `newTaskCommand`, etc., to handle JSON output.
>     - Adds JSON error handling in `task.go` and `instances.go`.
>   - **Testing**:
>     - Includes `demo-json-output.sh` script for testing JSON output.
>     - Updates `test_output.md` with test results and analysis.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=cline%2Fcline&utm_source=github&utm_medium=referral)<sup> for 5d583186b586bfd14345ba6f59efe54119fb3a54. You can [customize](https://app.ellipsis.dev/cline/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: cli/cmd/cline/main.go [hunk 1]
```diff
@@ -184,7 +210,80 @@ see the manual page: man cline`,
...
+	rootCmd.SilenceUsage = true
+
+	// Suppress Cobra's default error printing - we'll handle it ourselves
+	rootCmd.SilenceErrors = true
+	rootCmd.SilenceUsage = true
+
+	// Suppress Cobra's default error printing - we'll handle it ourselves
+	rootCmd.SilenceErrors = true
+	rootCmd.SilenceUsage = true
+
```
[the remark this answers is not in the document]
Comment (csells) [reply cli/cmd/cline/main.go]: fixed

File: cli/pkg/cli/output/json.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	"fmt"
+)
+
+// OutputJSONLine outputs a single JSON object as a line (JSONL format)
+func OutputJSONLine(obj map[string]interface{}) error {
+	jsonBytes, err := json.Marshal(obj)
+	if err != nil {
+		return fmt.Errorf("failed to marshal JSON: %w", err)
+	}
+	fmt.Println(string(jsonBytes))
```
[the remark this answers is not in the document]
Comment (csells) [reply cli/pkg/cli/output/json.go:14]: that's a pre-existing condition and not covered by this branch.

File: cli/pkg/cli/output/json.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+		"command": command,
+		"status":  "success",
+		"data":    data,
+	}
+	return OutputJSONLine(response)
+}
+
+// OutputCommandSuccess outputs a successful command result
+func OutputCommandSuccess(command string, data interface{}) error {
+	return OutputCommandStatus(command, "success", "", map[string]interface{}{"result": data})
```
[the remark this answers is not in the document]
Comment (csells) [reply cli/pkg/cli/output/json.go]: removed unused OutputCommandSuccess and changed command results from being output under "data" to "result".

File: cli/pkg/cli/version.go [hunk 4]
```diff
@@ -20,10 +21,27 @@ func NewVersionCommand() *cobra.Command {
 		RunE: func(cmd *cobra.Command, args []string) error {
 			// Versions are injected at build time via ldflags
 			if short {
+				// --short flag always outputs plain version, even in JSON mode
```
[the remark this answers is not in the document]
Comment (csells) [reply cli/pkg/cli/version.go:24]: Updated `cli/pkg/cli/version.go` flag description:

- Before: `"show only version number"`
- After: `"show only version number (outputs plain text, overrides --output-format)"`

File: cli/pkg/cli/task/manager.go [hunk 5]
```diff
@@ -562,12 +580,13 @@ func (m *Manager) ResumeTask(ctx context.Context, taskID string) error {
...
 	}
 
-	// This call handles cancellation of any active task
+	// This call handles cancellation of any active task and outputs success message
 	if err := m.ReinitExistingTaskFromId(ctx, taskID); err != nil {
 		return fmt.Errorf("failed to resume task %s: %w", taskID, err)
 	}
 
-	fmt.Printf("Task %s resumed successfully\n", taskID)
+	// Note: ReinitExistingTaskFromId already outputs success message in both JSON and plain formats
```
[the remark this answers is not in the document]
Comment (csells) [reply cli/pkg/cli/task/manager.go]: Updated such that:
- __Helper function__ (`ReinitExistingTaskFromId`): Silent, no output
- __Command code__ (`task open` in task.go): Handles all output for both JSON and plain/rich modes
- __Comments__: Accurately reflect the actual behavior

File: cli/e2e/format_validation_test.go [hunk 6]
```diff
@@ -0,0 +1,1037 @@
...
+				t.Error("missing command field")
+			}
+			if _, ok := response["status"]; !ok {
+				t.Error("missing status field")
+			}
+
+			// Success responses should have data
+			if response["status"] == "success" {
+				if _, ok := response["result"]; !ok {
+					t.Error("success response missing data field")
```
[the remark this answers is not in the document]
Comment (csells) [reply cli/e2e/format_validation_test.go:559]: accepted suggestion

File: cli/e2e/format_validation_test.go [hunk 7]
```diff
@@ -0,0 +1,1037 @@
...
+	setTempClineDir(t)
+
+	// Create instances
+	_ = mustRunCLI(ctx, t, "instance", "new")
+	out := mustRunCLI(ctx, t, "instance", "new", "--output-format", "json")
+	var newResponse map[string]interface{}
+	json.Unmarshal([]byte(out), &newResponse)
+	data, ok := newResponse["result"].(map[string]interface{})
+	if !ok {
+		t.Fatalf("failed to get data from instance new response")
```
[the remark this answers is not in the document]
Comment (csells) [reply cli/e2e/format_validation_test.go]: accepted suggestion

**Comment by dominiccooney** [comment]
#9021 replaced the Go CLI and removed the `instance`, `logs`, and `task` command architecture this PR changes; #10658 later deleted the remaining legacy `cli/` tree. The current TypeScript CLI provides `--json` NDJSON for headless automation and JSON output on its relevant batch commands, so I’m closing this as superseded.

Thank you for the substantial implementation and testing work here, and sorry we were slow to respond. We’d be glad to see future contributions; please open a focused issue if a current `apps/cli` command still lacks machine-readable output.
